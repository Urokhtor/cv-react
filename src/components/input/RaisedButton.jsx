import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './RaisedButton.scss';

export default class RaisedButton extends Component {

    render() {
        return (
            <div>
                <button onClick={this.props.onClick}>{this.props.text}</button>
            </div>
        );
    }
}

RaisedButton.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
};