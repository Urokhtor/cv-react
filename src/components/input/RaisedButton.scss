@import '../../styles/index.scss';

button {
  @extend .shadowed;
  position: relative;

  margin-top: $material-padding-normal;
  margin-bottom: $material-padding-normal;

  border: none;
  height: 36px;
  min-width: 88px;
  padding: 0 $material-padding-large;

  cursor: pointer;
  outline: none;

  font-size: 14pt;
  text-transform: uppercase;

  -webkit-border-radius: 2px;
  -moz-border-radius: 2px;
  border-radius: 2px;

  color: $colour-primary-text;
  background-color: $colour-accent;

}

button:active {
  @extend .shadowed-large;
  background-color: $colour-accent-dark;
}