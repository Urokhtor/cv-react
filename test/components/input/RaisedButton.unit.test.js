import React from 'react';
import RaisedButton from "../../../src/components/input/RaisedButton";

test('RaisedButton should match snapshot', (done) => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<RaisedButton text={'some-test-text'} onClick={mockOnClick}/>);

    expect(wrapper).toMatchSnapshot();

    done();
});

test('RaisedButton should not have executed callback if button has not been clicked', (done) => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<RaisedButton text={'some-test-text'} onClick={mockOnClick}/>);

    expect(mockOnClick).not.toHaveBeenCalled();

    done();
});

test('RaisedButton should execute callback when button is clicked', (done) => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<RaisedButton text={'some-test-text'} onClick={mockOnClick}/>);

    wrapper.find('button').simulate('click');

    expect(mockOnClick).toHaveBeenCalled();

    done();
});

test('RaisedButton should display text passed to it', (done) => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<RaisedButton text={'some-test-text'} onClick={mockOnClick}/>);

    expect(wrapper.find('button').text()).toEqual('some-test-text');

    done();
});