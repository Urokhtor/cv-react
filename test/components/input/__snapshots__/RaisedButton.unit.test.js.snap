// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaisedButton should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RaisedButton
    onClick={[MockFunction]}
    text="some-test-text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[MockFunction]}
      >
        some-test-text
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "some-test-text",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": "some-test-text",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[MockFunction]}
        >
          some-test-text
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some-test-text",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "some-test-text",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
